Backend/main.go

```diff
@@ -33,25 +32,38 @@ var (
 type User models.User
 
 func main() {
-	initialPostCount := 3
+	// getProfilePicture wont work without passing the db connection from main
+	// i tried putting the connection in the actual function instead of here and it will not work
+	// ...... i tried to make to work for like 1 hour
+	// im done -_-
+	dbName := os.Getenv("DB_NAME")
+	dbPass := os.Getenv("DB_PASS")
+	dbHost := os.Getenv("DB_HOST")
 
-	println("\n\n\nStarting time sort: \n")
+	dsn := fmt.Sprintf("%s:%s@tcp(%s:3306)/capstone", dbName, dbPass, dbHost)
 
-	testarr := feed.InitialFeedByTime(initialPostCount)
-	feed.DisplayModel(testarr)
-	testarr = feed.FeedByTime(initialPostCount)
-	feed.DisplayModel(testarr)
+	// Open a connection to the database
+	db, err := sql.Open("mysql", dsn)
+	if err != nil {
+		panic(err)
+	}
+	defer db.Close()
 
-	println("\n\n\nStarting random sort: \n")
+	// Ping the database to verify the connection is alive
+	err = db.Ping()
+	if err != nil {
+		panic(err)
+	}
+
+	http.HandleFunc("/upload", handlers.UploadImageHandler)
 
-	var used []int
-	testarr, used = feed.InitialFeedByRandom(3)
-	feed.DisplayModel(testarr)
-	testarr = feed.FeedByRandom(used)
-	feed.DisplayModel(testarr)
+	http.Handle("/uploads/", http.StripPrefix("/uploads/", http.FileServer(http.Dir("uploads"))))
 
-	// AddReaction("thumbs_up", 3, 3)
+	// Apply CORS middleware to the GetProfilePictureHandler
+	http.Handle("/getProfilePicture", enableCORS(http.HandlerFunc(handlers.GetProfilePictureHandler(db))))
 
+	log.Println("Server started on http://localhost:8080")
+	log.Fatal(http.ListenAndServe(":8080", nil))
```

Comment:
So the reason you were not able to run the app on 3000 the other day I think has something to do with this. Usually when we run main it starts the backend on 3000 but I think because you start a different port here it doesn't start the one on 3000 (not entirely sure). I think it makes sense to either have a separate file that will run the uploads API or more ideally the file uploads would be on port 3000 that way the entire API is all hosted on one server. I think you should be able to do it but you might know more based on the work you did on the profile pictures.